pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -486,7 +534,12 @@ func isSkipped(rprt *ResolvedPipelineRunTask, stateMap map[string]*ResolvedPipel
 	// if any of the parents have been skipped, skip as well
 	node := d.Nodes[rprt.PipelineTask.Name]
 	for _, p := range node.Prev {
-		skip := isSkipped(stateMap[p.Task.HashKey()], stateMap, d)
+		rcc := stateMap[p.Task.HashKey()]
```

Comment:
Is this something avoidable? marking a task skipped if one of the parents is failed or cancelled, this might interfere with `except` similar to `finally` but is invoked only when a task fails 🤔 .`Pipeline` is anyways `stopped` or `paused` (may be better term), resume it where we left off and report failure. Just brainstorming here 💭 how does these changes impact pipeline run status?

Reply:
Today, tasks are considered `skipped` (1) when its condition check fails and hence not executed and skipped (2) Or one of the parents is skipped. Tasks dependent on failed tasks are considered as `incomplete`.

Reply:
> Tasks are considered `skipped` (1) when its condition check fails and hence not executed and skipped (2) Or one of the parents is skipped. Tasks dependent on failed tasks are considered as `incomplete`.

Well, that's the logic I would like to change, i.e. recognize tasks that are going to be skipped as early as possible. This is useful when a pipeline is stopping, but it also gives a better indication to users about the actual incomplete. For instance finally tasks, would never go to skipped.

Reply:
> Is this something avoidable? marking a task skipped if one of the parents is failed or cancelled, this might interfere with `except` similar to `finally` but is invoked only when a task fails 🤔 .

Well, I'm not really marking the tasks, this computation is only used to decide what the overall status of the pipeline is, but not extra info is added into the pipeline state.

> `Pipeline` is anyways `stopped` or `paused` (may be better term), resume it where we left off and report failure. Just brainstorming here 💭 how does these changes impact pipeline run status?

I used stopping because I think it better reflects the fact that some tasks are still running. No new tasks are scheduled but something is still going on.
The pipeline status, as soon as a failure/task cancel occurs, is updated as follows:
- if other tasks are still running, the status remains "unknown", the reason is changed to "stopping"  and in the message "incomplete" tasks are counted as "skipped" as they will never be executed
- if no other task is running, the status is set to "failed", the reason is set to the appropriate one (failed or cancelled) and the message shows the final count of completed / failed / cancelled / skipped tasks

Reply:
But I think I should remove the check for "failed/cancelled" parent from the `isSkipped` anyways, since it does not serve any purpose right now (it did in my initial version of the code):
- if the pipeline is running normally (no failure, no cancel), this check will add anything
- if the pipeline is stopping, the actual number of skipped is given by:
```
		case !rprt.IsStarted() && isStopping:
			// If the pipeline is in stopping mode, all tasks that are not running
			// already will be skipped. Otherwise these tasks end up in the
			// incomplete count.
			skipTasks++
			withStatusTasks = append(withStatusTasks, rprt.PipelineTask.Name) 
```

Reply:
Ok, I pushed a new version that removed the changes to `isSkipped` but kept the unit test coverage for that. That way we can keep better track of the behaviour of that function independently of the `GetPipelineConditionStatus`

Reply:
> Is this something avoidable? 

Just to confirm, yes, it is avoidable, and I removed it now. Thank you!